### PR TITLE
Feature/soroban api improvements

### PR DESCRIPTION
**Pull request by piyalbasu**
Adding side-effect-less API methods in response to #830

### CODE REVIEW

File: @shared/api/external.ts [hunk 1]
```diff
@@ -132,3 +133,59 @@ export const requestConnectionStatus = async (): Promise<boolean> => {
...
+  try {
+    response = await sendMessageToContentScript({
+      type: EXTERNAL_SERVICE_TYPES.REQUEST_ALLOWED_STATUS,
+    });
+  } catch (e) {
+    console.error(e);
+  }
+
+  return response.isAllowed;
+};
```
Comment (piyalbasu) [inline @shared/api/external.ts:151]: check to see if the dapp is currently on the allowlist

File: @shared/api/external.ts [hunk 2]
```diff
@@ -132,3 +133,59 @@ export const requestConnectionStatus = async (): Promise<boolean> => {
...
+    error: "",
+  };
+
+  try {
+    response = await sendMessageToContentScript({
+      type: EXTERNAL_SERVICE_TYPES.SET_ALLOWED_STATUS,
+    });
+  } catch (e) {
+    console.error(e);
+  }
```
Comment (piyalbasu) [inline @shared/api/external.ts:165]: put the current dapp on the allowlist

File: @shared/api/external.ts [hunk 3]
```diff
@@ -132,3 +133,59 @@ export const requestConnectionStatus = async (): Promise<boolean> => {
...
+
+  const { isAllowed, error } = response;
+
+  if (error) {
+    throw error;
+  }
+  return isAllowed;
+};
+
+export const requestUserInfo = async (): Promise<UserInfo> => {
```
Comment (piyalbasu) [inline @shared/api/external.ts:175]: get userinfo (includes public key) in a way that doesn't prompt the user. It just returns an empty string if the dapp isn't allowed or the user needs to re-auth

**Review by aristidesstaffieri — APPROVED** [review]
nice lgtm

**Pull request merged**